File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-1.cpp
class Solution {
public:
    int characterReplacement(string s, int k) {
        vector<int> counts(26,0);
        int l = 0;
        int maxcount = 0;
        int maxlen = 0;
        int n = s.size();
        for(int i = 0;i<n;i++){
            counts[s[i]-'A']++;
            maxcount = max(maxcount,counts[s[i]-'A']);
            while((i-l+1) - maxcount > k){
                counts[s[l]-'A']--;
                l++;
            }
            maxlen=max(maxlen,i-l+1);
        }
        return maxlen;
    }
};

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-1.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        n = len(prices)
        if n == 1:
            return 0
        l,r = 0,1
        val = 0
        while r < n:
            val  = max(val,prices[r] - prices[l])
            if prices[r] < prices[l]:
                l = r
                r+=1
            else:
                r+=1
            
        return val